services:
  redis-test:
    image: redis:alpine
    container_name: redis-test
    ports:
      - "6379:6379"
    networks:
      - test-net
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 5s
      timeout: 5s
      retries: 5

  test-db:
    image: postgres:14-alpine
    container_name: test-db
    environment:
      POSTGRES_DB: test_contacts_db
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    ports:
      - "5433:5432"
    networks:
      - test-net
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

  contacts-test:
    build:
      context: .
      dockerfile: Dockerfile.test
    volumes:
      - .:/app
    container_name: contacts-test
    depends_on:
      test-db:
        condition: service_healthy
      redis-test:
        condition: service_healthy
    environment:
      TEST_ENV: "1"
      DOTENV_PATH: /app/.env.test
      PYTHONPATH: /app/src
      TZ: Europe/Kyiv
    networks:
      - test-net
    restart: "no"
    command: >
      sh -c "pytest --cov=src --tb=short; tail -f /dev/null"

networks:
  test-net:
    driver: bridge
